product/scanning.py

```diff
@@ -117,26 +119,40 @@ def retrieve_scan_result(
         product = self._api.scan_result(product_id=product_id)
         return product
 
+    def supported_modes(self):
+        return [ProcessingMode.UPLOAD_IF_CHANGED, ProcessingMode.RESCAN]
+
     def _determine_upload_action(
             self,
             container_image: ContainerImage,
             scan_result: AnalysisResult,
     ):
-        if self._processing_mode is not ProcessingMode.UPLOAD_IF_CHANGED:
+        if self._processing_mode not in self.supported_modes():
```

Comment:
that is not a good idea. actually, this is kind of even worse than not checking the modi at all.